R/onLoad.R

```diff
@@ -27,53 +27,60 @@
 #############################################################################
 
 .onLoad <- function(libname, pkgname) {
+  
+  javaCheck <- function() {
+    # Java version check, without .jinit (we do .jpackage after downloading resources)
+    rawVersion <- system2("java", c("-version"), stdout = TRUE, stderr = TRUE)
+    jv <- regmatches(rawVersion[1], regexpr("[0-9]+\\.[0-9\\.]*", rawVersion[1]))
+    twoFirst <- substr(jv, 1L, 2L)
+    if(twoFirst == "1.") {
+      jvn <- as.numeric(substr(jv,3L,3L))
+    } else {
+      jvn <- as.numeric(twoFirst)
+    }
+    if (jvn<8 || jvn>15) stop(paste0("Installed java version ",jv ," is not between Java>=8 and <=15! This is needed for this package"))
```

Comment:
I would probably change this message slightly to something like "XLConnect requires Java >= 8 and <= 15. Detected version of Java is X".